mobile/src/reducers/calendar/calendar.epics.ts

```diff
@@ -1,19 +1,30 @@
-import { UserActions, LoadUserEmployeeFinished } from '../user/user.action';
+import { LoadUserEmployeeFinished } from '../user/user.action';
 import { ActionsObservable } from 'redux-observable';
 import { Observable } from 'rxjs/Observable';
-import { ajaxGetJSON } from 'rxjs/observable/dom/AjaxObservable';
+import { ajaxGetJSON, ajaxPost } from 'rxjs/observable/dom/AjaxObservable';
 import { apiUrl } from '../const';
 import { deserializeArray } from 'santee-dcts';
 import { loadCalendarEventsFinished } from './calendar.action';
 import { loadFailedError } from '../errors/errors.action';
-import { CalendarEvents, CalendarEventsType, CalendarEventStatus, DatesInterval } from './calendar-events.model';
-import moment from 'moment';
+import { CalendarEvents } from './calendar-events.model';
+import { ConfirmClaimSickLeave, sickLeaveSaved, SickLeaveSaved } from './sick-leave.action';
 
-export const loadCalendarEventsFinishedEpic$ = (action$: ActionsObservable<LoadUserEmployeeFinished>) =>
-    action$.ofType('LOAD-USER-EMPLOYEE-FINISHED')
+export const loadCalendarEventsFinishedEpic$ = (action$: ActionsObservable<LoadUserEmployeeFinished | SickLeaveSaved>) =>
+    action$.filter(x => x.type === 'LOAD-USER-EMPLOYEE-FINISHED' || x.type === 'SICK-LEAVE-SAVED')
         .switchMap(x =>
             ajaxGetJSON(`${apiUrl}/employees/${x.employee.employeeId}/events`).map((obj: Object[]) => deserializeArray(obj, CalendarEvents))
         )
         .map(x => loadCalendarEventsFinished(x))
         .catch((e: Error) => Observable.of(loadFailedError(e.message)));
 
+export const calendarEventsSavedEpic$ = (action$: ActionsObservable<ConfirmClaimSickLeave>) =>
+    action$.ofType('CONFIRM-CLAIM-SICK-LEAVE')
+        .switchMap(x => {
+
+            return ajaxPost(
+                `${apiUrl}/employees/${x.employee.employeeId}/events`,
+                x.calendarEvents,
+                { 'Content-Type': 'application/json' }).map(() => x.employee);
```

Comment:
this method returns you a newly created event. Why do you not use it? I think you'll need its id, for example.

Reply:
also, the whole switch method probably needs to be moved to another funciton, or used in a pipe